src/Robo/Common/ComposerJson.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Acquia\Blt\Robo\Common;
+
+use Acquia\Blt\Robo\Exceptions\BltException;
+
+/**
+ * Class ComposerWriter.
+ *
+ * @package Acquia\Blt\Robo\Common
+ *
+ * Reads and writes to composer.json in a friendly and consistent way.
+ */
+class ComposerJson {
+
+  /**
+   * Filepath.
+   *
+   * @var string
+   */
+  private $filepath;
+
+  /**
+   * Composer.json contents.
+   *
+   * @var array
+   */
+  public $contents;
+
+  /**
+   * ComposerWriter constructor.
+   *
+   * @param string $directory
+   *   Directory holding composer.json.
+   * @param string $filename
+   *   Typically composer.json.
+   *
+   * @throws \Acquia\Blt\Robo\Exceptions\BltException
+   */
+  public function __construct(string $directory, string $filename = 'composer.json') {
+    $this->filepath = $directory . DIRECTORY_SEPARATOR . $filename;
+    if (!file_exists($this->filepath)) {
+      throw new BltException("Could not find composer.json");
+    }
+    $this->contents = json_decode(file_get_contents($this->filepath), TRUE);
+  }
+
+  /**
+   * Writes contents to file.
+   */
+  public function write() {
```

Comment:
There's some logic here that deserves a unit test to make sure we understand requirements in the future.